src/TBuffer.cpp

```diff
@@ -2611,59 +2587,60 @@ void TBuffer::logRemainingOutput()
     mpHost->mpConsole->mLogStream.flush();
 }
 
-// returns how many new lines have been inserted by the wrapping action
-int TBuffer::wrapLine(int startLine, int screenWidth, int indentSize, TChar& format)
+// The difference between this and wrap is, wrap does it from startLine all the way to the end
+// while wrapLine only does it for one line.
+// Thus, wrap uses pop_back and push_back while wrapLine uses remove_at and insert.
+// Returns how many new lines have been added by the wrapping action
+inline int TBuffer::wrapLine(int startLine, int screenWidth, int indentSize, TChar& format, bool onlyWrapOneLine)
 {
-    if (startLine < 0) {
-        return 0;
-    }
-    if (static_cast<int>(buffer.size()) <= startLine) {
+    if (static_cast<int>(buffer.size()) < startLine || startLine < 0) {
         return 0;
     }
+
     std::queue<std::deque<TChar>> queue;
     QStringList tempList;
+    QStringList timeList;
+    QList<bool> promptList;
     int lineCount = 0;
 
-    for (int line = startLine, total = static_cast<int>(buffer.size()); line < total; ++line) {
-        if (line > startLine) {
+    const auto hostFont = mpHost->getDisplayFont();
+    
+    // Loop from current line up to the last line sitting in buffer
+    for (int i = startLine, total = static_cast<int>(buffer.size()); i < total; ++i) {
+        if (onlyWrapOneLine && i > startLine) {
             break; //only wrap one line of text
         }
+        // Capture isPrompt
+        bool isPrompt = promptBuffer.at(i);
         std::deque<TChar> newLine;
-        QString lineText;
+        QString lineText = "";
```

Comment:
```suggestion
        QString lineText;
```
Does this also work? QString starts empty by default